imports/client/components/SplitPanePlus.tsx

```diff
@@ -111,168 +154,175 @@ class SplitPanePlus extends React.Component<SplitPanePlusProps, SplitPanePlusSta
-    const halfResizerSize = this.measure(this.resizerNode()!) / 2;
-    let autoCollapsePrimary = this.props.autoCollapse1!;
-    let autoCollapseSecondary = this.props.autoCollapse2!;
+    const fullSize = this.measure(this.splitPaneNode());
+    let autoCollapsePrimary = this.props.autoCollapse1;
+    let autoCollapseSecondary = this.props.autoCollapse2;
     if (this.props.primary !== 'first') {
-      autoCollapsePrimary = this.props.autoCollapse2!;
-      autoCollapseSecondary = this.props.autoCollapse1!;
+      autoCollapsePrimary = this.props.autoCollapse2;
+      autoCollapseSecondary = this.props.autoCollapse1;
     }
-    if (size + halfResizerSize <= autoCollapsePrimary) {
-      // Collapse Primary
+    if (autoCollapsePrimary >= 0 && size <= autoCollapsePrimary) {
       return this.props.primary === 'first' ? 1 : 2;
-    } else if (size + halfResizerSize >= fullSize - autoCollapseSecondary) {
-      // Collapse Secondary
+    } else if (autoCollapseSecondary >= 0 && size >= fullSize - autoCollapseSecondary) {
       return this.props.primary === 'first' ? 2 : 1;
     }
     return 0;
   }
 
   splitPaneNode(): HTMLElement | null {
-    if (!this.ref.current) {
-      return null;
-    }
-    if (!(this.ref.current.firstChild instanceof HTMLElement)) {
+    if (!this.ref.current || !(this.ref.current.firstChild instanceof HTMLElement)) {
       return null;
     }
     return this.ref.current.firstChild;
   }
 
-  primaryPaneNode() {
+  primaryPaneNode(): HTMLElement | null {
     const root = this.splitPaneNode();
     const className = `Pane${this.props.primary === 'first' ? 1 : 2}`;
     return root ? _.find(root.childNodes, (n) => (
       n instanceof Element &&
       n.classList.contains(className)
-    )) as Element : null;
+    )) as HTMLElement : null;
   }
 
-  secondaryPaneNode() {
+  secondaryPaneNode(): HTMLElement | null {
     const root = this.splitPaneNode();
     const className = `Pane${this.props.primary === 'first' ? 2 : 1}`;
     return root ? _.find(root.childNodes, (n) => (
       n instanceof Element &&
       n.classList.contains(className)
-    )) as Element : null;
+    )) as HTMLElement : null;
   }
 
-  resizerNode() {
+  resizerNode(): HTMLElement | null {
     const root = this.splitPaneNode();
     return root ? _.find(root.childNodes, (n) => (
       n instanceof Element &&
       n.classList.contains('Resizer')
-    )) as Element : null;
+    )) as HTMLElement : null;
   }
 
-  measure(elem: Element) {
+  measure(elem: HTMLElement | null): number {
     if (!elem) {
       return NaN;
     }
     return this.props.split === 'vertical' ? elem.clientWidth : elem.clientHeight;
   }
 
   render() {
-    const paneProps: SplitPaneProps = _.extend({}, this.props);
-
-    const defaultPaneStyles = { overflow: 'auto' };
-    const maxRangeStyles = (this.props.split === 'vertical' ?
-      { maxWidth: '100%' } :
-      { maxHeight: '100%' }
+    const paneProps: { [key: string]: any; } = _.pick(
```

Comment:
Can you declare this as, like, `Partial<SplitPaneProps>`?

Reply:
I thought so too and wasted a lot of time last night trying to make it work, but for some reason pane1ClassName and pane2ClassName (which clearly work, because I'm using them) throw errors as not existing. For now, I just added them to SplitPaneProps, but that's obviously not great. Any ideas?

Reply:
It looks like they're just missing from the type declarations in upstream: https://github.com/tomkp/react-split-pane/blob/master/index.d.ts#L7-L28. Probably worth filing a bug over at some point

Reply:
Well I feel like a fool.  I glanced at https://github.com/tomkp/react-split-pane/blob/master/src/SplitPane.js#L352-L378, decided it was in there, and never again considered the possibility this was just a bug on that side. Until it's fixed, as we ok with the patched in approach I used?

Reply:
Yeah, react-split-pane isn't actually written in Typescript, so they have the type definitions separate from the code...which means they can be wrong